### PR TITLE
Fix/exit once the scan has completed

### PR DESCRIPTION
**Pull request by elyousfi5**
[Screencast from 2024-08-15 16-32-54.webm](https://github.com/user-attachments/assets/d1cc82f0-895d-49f0-a96c-b6b76a78523c)

### CODE REVIEW

File: tests/runtimes/local/runtime_test.py [hunk 1]
```diff
@@ -332,3 +333,34 @@ def testRuntimeScanList_whenDockerIsDown_DoesNotCrash(
...
+    agent_group_definition = definitions.AgentGroupDefinition(
+        agents=[definitions.AgentSettings(key="agent/ostorlab/agent42")]
+    )
+
+    local_runtime_instance.scan(
+        title="test local",
+        agent_group_definition=agent_group_definition,
+        assets=[android_apk.AndroidApk(content=b"APK")],
+    )
+    sleep(1)
```
Comment (ohachimOs) [inline tests/runtimes/local/runtime_test.py:362]: Why the sleep?

Comment (elyousfi5) [reply tests/runtimes/local/runtime_test.py:362]: to give the thread sometime to check the services and exit

**Review by ohachimOs — DISMISSED** [review]

File: src/ostorlab/runtimes/local/log_streamer.py [hunk 2]
```diff
@@ -40,6 +40,7 @@ class LogStream:
     def __init__(self):
         self._threads = []
         self._color_map = {}
+        self.active_services = []
```
Comment (3asm) [inline src/ostorlab/runtimes/local/log_streamer.py:43]:
```suggestion
        self._services = []
```

File: src/ostorlab/runtimes/local/runtime.py [hunk 3]
```diff
@@ -232,6 +234,10 @@ def scan(
             console.info("Updating scan status")
             self._update_scan_progress("IN_PROGRESS")
             console.success("Scan created successfully")
+            scan_complete_check = threading.Thread(
```
Comment (3asm) [inline src/ostorlab/runtimes/local/runtime.py:237]:
```suggestion
            scan_complete_thread = threading.Thread(
```

File: src/ostorlab/runtimes/local/runtime.py [hunk 4]
```diff
@@ -254,6 +260,23 @@ def scan(
...
+
+        stop_event = threading.Event()
+        while stop_event.is_set() is False:
+            for service_id in list(self._log_streamer.active_services):
+                try:
+                    self._docker_client.services.get(service_id)
+                except docker_errors.NotFound:
+                    self._log_streamer.active_services.remove(service_id)
+            if len(self._log_streamer.active_services) == 0:
+                console.success("Scan completed.")
```
Comment (3asm) [inline src/ostorlab/runtimes/local/runtime.py:276]:
```suggestion
                console.success("Scan done.")
```

**Review by 3asm — CHANGES_REQUESTED** [review]

**Review by 3asm — APPROVED** [review]

**Pull request merged**